examples/examples_test.go

```diff
@@ -0,0 +1,218 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package examples_test
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("Example YAML Validation", func() {
+	const namespace = "openperouter-system"
+
+	BeforeEach(func() {
+		// Create the openperouter-system namespace for testing
+		ns := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespace,
+			},
+		}
+		err := k8sClient.Create(context.Background(), ns)
+		if err != nil && !strings.Contains(err.Error(), "already exists") {
+			Expect(err).NotTo(HaveOccurred())
+		}
+	})
+
+	AfterEach(func() {
+		// Clean up all resources after each test
+		cleanupResources(namespace)
+	})
+
+	// Dynamically discover and test all openpe.yaml files in the examples directory
+	exampleFiles := discoverExampleFiles()
+	for _, exampleFile := range exampleFiles {
+		// Capture the variable for the closure
+		file := exampleFile
+
+		Context(file.testName, func() {
+			It("should successfully create resources from "+file.relativePath, func() {
+				validateExampleFile(file.relativePath, namespace)
+			})
+		})
+	}
+})
+
+// exampleFile represents a discovered example YAML file
+type exampleFile struct {
+	relativePath string
+	testName     string
+}
+
+// discoverExampleFiles walks the examples directory and finds all YAML files
+// containing OpenPERouter CRs (openpe.yaml and cluster-*-openpe.yaml files)
+func discoverExampleFiles() []exampleFile {
+	var files []exampleFile
+	examplesDir := "."
+
+	err := filepath.Walk(examplesDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// Skip the examples directory itself and non openpe YAML files
+		if info.IsDir() || (!strings.HasSuffix(info.Name(), "openpe.yaml") && !strings.HasSuffix(info.Name(), "openpe.yml")) {
```

Comment:
I think it'd be better to provide a list of CRD names we want to validate, rather than going over the name